proto/ignition/msgs/dataframe.proto

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+syntax = "proto3";
+package ignition.msgs;
+option java_package = "com.ignition.msgs";
+option java_outer_classname = "DataframeProtos";
+
+/// \ingroup ignition.msgs
+/// \interface Dataframe
+/// \brief A message containing some payload and who are the sender and
+/// destination.
+
+import "ignition/msgs/header.proto";
+
+message Dataframe
+{
+  /// \brief Header data.
+  Header header      = 1;
+
```

Comment:
While a header with the correct time stamp and the source and destination addresses are sufficient for the MBZIRC use case, we should also log the position of where the message is starting from this is useful for the MBARI use case. The reason is that the message take in the order of a few seconds to propagate hence it would help if the original position were logged as the vehicle may have moved in this time frame.

Reply:
Does it make sense to store that information in the specific comms model during the 1st step?

Reply:
That's one possibility, but it would lead to very complicated code. We probably can revisit this afterwards.